### PR TITLE
[ELY-302] SHA-512-256 digest algorithm and SASL mech

### PR DESCRIPTION
**Pull request by hkalina**
Based on https://github.com/wildfly-security/wildfly-elytron/pull/990

https://issues.jboss.org/browse/ELY-302

### CODE REVIEW

File: src/main/java/org/wildfly/security/digest/Sha512_256MessageDigest.java [hunk 1]
```diff
@@ -0,0 +1,192 @@
+package org.wildfly.security.digest;
```
Comment (fjuma) [inline src/main/java/org/wildfly/security/digest/Sha512_256MessageDigest.java]: Missing a copyright header.

Comment (honza889) [reply src/main/java/org/wildfly/security/digest/Sha512_256MessageDigest.java]: Fixed.

File: src/main/java/org/wildfly/security/http/HttpConstants.java [hunk 2]
```diff
@@ -45,6 +45,7 @@ private HttpConstants() {
     private static final String CONFIG_BASE = HttpConstants.class.getPackage().getName();
     public static final String CONFIG_CONTEXT_PATH = CONFIG_BASE + ".context-path";
     public static final String CONFIG_REALM = CONFIG_BASE + ".realm";
+    public static final String CONFIG_ALGORITHM = CONFIG_BASE + ".algorithm";
```
Comment (darranl) [inline src/main/java/org/wildfly/security/http/HttpConstants.java]: Is this used anywhere?  I think this was part of a prior version of this PR before it evolved further?

**Comment by darranl** [comment]
Other than my comment about the config parameter I think this is almost ready to merge.

**Comment by hkalina** [comment]
@darranl fixed.

**Comment by fjuma** [comment]
I've added a +1 but held off on merging since the PR that this is based on (#990) should be merged first.

**Pull request merged**